packages/plugin-node-uncaught-exception/uncaught-exception.js

```diff
@@ -12,12 +12,15 @@ module.exports = {
         unhandled: true,
         severityReason: { type: 'unhandledException' }
       }, 'uncaughtException handler', 1)
-      c._notify(event, () => {}, (e, event) => {
-        if (e) c._logger.error('Failed to send event to Bugsnag')
-        c._config.onUncaughtException(err, event, c._logger)
+      return new Promise(resolve => {
+        c._notify(event, () => {}, (e, event) => {
+          if (e) c._logger.error('Failed to send event to Bugsnag')
+          c._config.onUncaughtException(err, event, c._logger)
+          resolve()
+        })
       })
     }
-    process.on('uncaughtException', _handler)
+    process.prependListener('uncaughtException', _handler)
```

Comment:
This is a breaking change for the node notifier so we might want to avoid this change by creating a separate uncaught-exception plugin just for aws lambda. Thoughts?

Reply:
I'm not sure why it'd be breaking; do you just mean that `prependListener` isn't available on old Node versions?

In the unhandled rejection plugin we have a guard for this:

https://github.com/bugsnag/bugsnag-js/blob/aced3db494a4a779cb2911dbfbdd9e6c529c807c/packages/plugin-node-unhandled-rejection/unhandled-rejection.js#L21-L26

Reply:
I was thinking if a user's application code or other third-party library also adds listeners that this change could alter the order and break some implicit assumptions

Reply:
I'd say that's pretty unlikely and as we've already made this change in the unhandled rejection plugin, it'd be OK to do here as well

We could postpone this change until a v8.0.0 though, since it should be very rare — IIUC this is only really relevant if you have top-level code that throws? I guess also using regular callbacks, but doing that without waiting for the result is likely to cause other issues in a Lambda function